Foundation/NSPathUtilities.swift

```diff
@@ -574,6 +574,18 @@ extension FileManager {
 }
 
 public func NSSearchPathForDirectoriesInDomains(_ directory: FileManager.SearchPathDirectory, _ domainMask: FileManager.SearchPathDomainMask, _ expandTilde: Bool) -> [String] {
+    let bundle = Bundle.main
+    let bundlePath = bundle.bundlePath
+    if domainMask == .userDomainMask {
+        switch directory {
+        case .libraryDirectory:
+            return [bundlePath + "/Library"]
+        case .documentDirectory:
+            return [bundlePath + "/Documents"]
+        default:
+            break
+        }
+    }
```

Comment:
In iOS this function returns Library or Documents directory inside the main bundle

Sent from my iPhone

> On Aug 31, 2016, at 9:09 AM, Tony Parker <notifications@github.com> wrote:
> 
> In Foundation/NSPathUtilities.swift:
> 
> > @@ -574,6 +574,18 @@ extension FileManager {
> >  }
> >  
> >  public func NSSearchPathForDirectoriesInDomains(_ directory: FileManager.SearchPathDirectory, _ domainMask: FileManager.SearchPathDomainMask, _ expandTilde: Bool) -> [String] {
> > +    let bundle = Bundle.main
> > +    let bundlePath = bundle.bundlePath
> > +    if domainMask == .userDomainMask {
> > +        switch directory {
> > +        case .libraryDirectory:
> > +            return [bundlePath + "/Library"]
> > +        case .documentDirectory:
> > +            return [bundlePath + "/Documents"]
> > +        default:
> > +            break
> > +        }
> > +    }
> Shouldn't the user domain be in the home directory (~)? The search functionality on Bundle itself is used for finding resources inside the main bundle; and bundles typically do not have a Library and Documents directory inside them.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>